docs/src/rfc/text/0000-patina-boot.md

```diff
@@ -0,0 +1,640 @@
+
+### Monolithic Boot Manager Component
+
+Single comprehensive component handling console initialization, boot discovery, and orchestration with
+configuration options.
+
+**Limitation**: Individual pieces cannot be replaced independently of the reference implementation.
+
+### Library-Only Approach
+
+Library functions only (`parse_boot_variables()`, `discover_console_devices()`, etc.) with no reference
+components. Platforms implement orchestration.
+
+**Limitation**: Code duplication across platforms. Simple platforms implement significant boilerplate.
+Higher adoption barrier.
+
+### Chosen Approach: Modular Components with Platform Orchestration
+
+Three reference components (`ConsoleDiscovery`, `BootOptionDiscovery`, `BootOrchestrator`) that platforms
+use as-is, compose selectively, or replace entirely. Platform orchestrators control execution order and
+flow.
+
+**Benefits**: Zero custom code for simple platforms. Custom orchestration support (change order, skip
+steps, custom control flow). Independent component replacement. Platforms carry only used code. Leverages
+Patina's component model.
+
+## Design Assumptions
+
+This RFC's architecture is based on the following design considerations:
+
+1. **Platform diversity**: Platforms range from simple embedded systems with fixed boot paths to complex
+   servers with network boot, remote attestation, and custom boot policies
+2. **Customization requirement**: Platform-specific boot flows, connection policies, and orchestration
+   requirements necessitate composable components rather than monolithic implementations
+3. **Progressive disclosure**: Architecture supports both simple use cases (single composite component)
+   and complex scenarios (individual component composition with custom orchestration)
+4. **Scope**: This RFC covers console initialization alongside boot functionality because both are
+   required UEFI Boot Manager responsibilities. ConsoleDiscovery is optional for headless systems but
+   included for completeness
+
+## StandardBootManager Component
+
+`StandardBootManager` is a composite component that provides complete UEFI-compliant boot functionality
+with minimal configuration. It internally uses ConsoleDiscovery, BootOptionDiscovery, and BootOrchestrator,
+and orchestrates driver dispatch for complete device topology enumeration.
+
+**Design approach:** Progressive disclosure - simple platforms use one component with defaults, complex
+platforms customize via closures/traits or compose individual components directly. See "Guide-Level
+Explanation" for usage examples.
+
+## Rust Code Design
+
+### Component Architecture
+
+The boot components provide three independent implementations that map to UEFI specification boot phases.
+These components transition firmware from initialization to OS execution.
+
+#### Component Decomposition
+
+The three components map to distinct phases of UEFI boot. **Console initialization** populates console
```

Comment:
Maybe some commentary on why this three-way decomposition is chosen. Other decompositions could be imagined with more steps (for example, you could break out **System Bus Initialization** from **Console initialization** and stick PCI enumeration in there). I'm not saying the three-way is wrong, necessarily, but would be useful to have a rationale for why that particular decomposition is preferred.